contracts/Overtime/SportsAMMV2.sol

```diff
@@ -445,61 +509,91 @@ contract SportsAMMV2 is Initializable, ProxyOwned, ProxyPausable, ProxyReentranc
     function _handleDifferentCollateral(
         uint _buyInAmount,
         address _collateral,
-        bool _isEth,
-        address _fromAddress
-    ) internal nonReentrant notPaused {
-        require(multicollateralEnabled, "Multi collateral not enabled");
-        uint collateralQuote = multiCollateralOnOffRamp.getMinimumNeeded(_collateral, _buyInAmount);
-
+        address _fromAddress,
+        bool _isEth
+    ) internal returns (address lqPool, uint collateralPrice) {
+        require(multicollateralEnabled, "Multi-collat not enabled");
         uint exactReceived;
+        lqPool = collateralPool[_collateral];
+        if (lqPool != address(0)) {
+            if (_collateral == multiCollateralOnOffRamp.WETH9() && _isEth) {
+                // WETH specific case
+                require(msg.value >= _buyInAmount, "Insuff ETH sent");
+                uint balanceBefore = IERC20(_collateral).balanceOf(address(this));
+                ICollateralUtility(_collateral).deposit{value: msg.value}();
+                uint balanceDiff = IERC20(_collateral).balanceOf(address(this)) - balanceBefore;
+                require(balanceDiff == msg.value, "Insuff WETH");
+                exactReceived = balanceDiff;
+            } else {
+                // Generic case for any collateral used (THALES/ARB/OP)
+                IERC20(_collateral).safeTransferFrom(msg.sender, address(this), _buyInAmount);
+                exactReceived = _buyInAmount;
+            }
 
-        if (_isEth) {
-            require(_collateral == multiCollateralOnOffRamp.WETH9(), "Wrong collateral sent");
-            require(msg.value >= collateralQuote, "Not enough ETH sent");
-            exactReceived = multiCollateralOnOffRamp.onrampWithEth{value: msg.value}(msg.value);
+            collateralPrice = ICollateralUtility(ICollateralUtility(address(multiCollateralOnOffRamp)).priceFeed())
```

Comment:
ensuring collateralPRice is > 0 might make sense

Reply:
done